contracts/BC_fusion/AirDrop.sol

```diff
@@ -0,0 +1,128 @@
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+import "./interface/ITokenHub.sol";
+import "./interface/IAirDrop.sol";
+import "./System.sol";
+import "./lib/Utils.sol";
+
+contract AirDrop is IAirDrop, ReentrancyGuardUpgradeable, System {
+    string public constant sourceChainID = "Binance-Chain-Ganges";
+    address public approvalAddress = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa;
+    bytes32 public merkleRoot = 0x0000000000000000000000000000000000000000000000000000000000000000;
```

Comment:
can you check the `stakehub` contract, there is `assetProtector` and `blacklist`, I wish airdrop can also pause  the contract and add blacklist as well

Reply:
added